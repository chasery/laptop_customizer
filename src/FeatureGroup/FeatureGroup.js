import React from "react";
import FeatureItem from "../FeatureItem/FeatureItem";
import slugify from "slugify";
import "./FeatureGroup.css";

class FeatureGroup extends React.Component {
  render() {
    const options = this.props.featureOptions.map((option) => {
      const { feature, selected, updateFeature, USCurrencyFormat } = this.props;
      const itemHash = slugify(JSON.stringify(option));

      return (
        <FeatureItem
          key={itemHash}
          feature={feature}
          item={option}
          itemHash={itemHash}
          selected={selected}
          updateFeature={updateFeature}
        />
      );
    });

    return (
      <fieldset className="feature">
        <legend className="feature__name">
          <h3>{this.props.feature}</h3>
        </legend>
        {options}
      </fieldset>
    );
  }
}

FeatureGroup.defaultProps = {
  featureOptions: [],
  selected: [],
};

export default FeatureGroup;
