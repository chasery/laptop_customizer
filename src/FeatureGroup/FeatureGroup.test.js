import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FeatureGroup from "./FeatureGroup";

describe("FeatureGroup component", () => {
  it("renders without errors", () => {
    const wrapper = shallow(<FeatureGroup />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
