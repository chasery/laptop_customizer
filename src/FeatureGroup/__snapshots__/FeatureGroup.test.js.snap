// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureGroup component renders without errors 1`] = `
<fieldset
  className="feature"
>
  <legend
    className="feature__name"
  >
    <h3 />
  </legend>
</fieldset>
`;
